xrayvision/tests/test_clean.py

```diff
@@ -41,6 +43,65 @@ def test_hogbom_intensity_change(self):
         final_image = np.add(clean.dirty_map, clean.point_source_map)
         assert np.allclose(final_image, clean_map)
 
+    def test_clean_usecase(self):
+        N = M = 64
+        pos1 = [15, 30]
+        pos2 = [40, 32]
+        # Creating a "clean" map as a base with 2 point sources
+        clean_map = np.zeros((N, M))
+        clean_map[pos1[0], pos1[1]] = 1.
+        clean_map[pos2[0], pos2[1]] = 0.8
+
+        u, v = np.meshgrid(np.arange(N), np.arange(M))
+        uv_in = np.array([u, v]).reshape(2, N * M)
+        vis_in = np.zeros(N * M, dtype=complex)
+
+        vis = Visibility(uv_in, vis_in)
+        vist = vis.from_map(clean_map)
+        vis.vis = vist
+
+        random.seed(0)
+
+        indexes = []
+
+        for i in range(N*M):
+            delete_it = random.choice([True, False])
+            if delete_it:
+                indexes.append(i)
+
+        vis.vis = np.delete(vis.vis, indexes)
+        vis.uv = np.delete(vis.uv, indexes, 1)
+
+        dirty_map = np.zeros((N, M))
+        dirty_map = vis.to_map(dirty_map)
+
+        save_vis = copy.deepcopy(vis.vis)
+
+        vis.vis = np.zeros(vis.vis.shape)
+        input_delta = np.zeros((N, M))
+        input_delta[N//2, M//2] = 1.
+        vis.from_map(input_delta)
+
+        dirty_beam = np.zeros((N, M))
+        dirty_beam = vis.to_map(dirty_beam)
+
+        vis.vis = save_vis
+        clean = Hogbom(vis, dirty_beam, 1e-14, (N, M), gain=1.0)
+        while not clean.iterate():
+            pass
+        final_image = np.add(clean.dirty_map, clean.point_source_map)
+        assert abs(1. - final_image[pos1[0], pos1[1]]) < 1e-1
```

Comment:
It possible there could be other bright spots in the image it would be better to find these (see L130+) and then compare their:
* locations
* brightness
the number you have for the tolerance seem a bit arbitrary why are they not 1e-14

Reply:
I no longer check the difference between the calculated and the actual value in terms of magntitude. I just check if there was a succesful amplification or not. Because of already pointed out "issues" (zero padding of the shifted dirty beam) it can not reach the expected value without giving a right gain just for that (and because the dirty beam's size is equal with the size of the dirty map). This is not a real life scenario, you do not know the expected value for gain beforehand, because if you would know you would not need CLEAN.